signal/signal.go

```diff
@@ -34,20 +34,24 @@ type Interceptor struct {
 
 	// quit is closed when instructing the main interrupt handler to exit.
 	quit chan struct{}
+
+	// exitCode is the exit code status when the main interrupt handler exits
+	exitCode chan int
 }
 
 // Intercept starts the interception of interrupt signals and returns an `Interceptor` instance.
 // Note that any previous active interceptor must be stopped before a new one can be created
-func Intercept() (Interceptor, error) {
+func Intercept() (*Interceptor, error) {
 	if !atomic.CompareAndSwapInt32(&started, 0, 1) {
-		return Interceptor{}, errors.New("intercept already started")
+		return &Interceptor{}, errors.New("intercept already started")
```

Comment:
can just return nil here, safe to assume that it won't be used when there's a non-nil error :)